test/gtest/ucs/test_memtype_cache.cc

```diff
@@ -337,19 +356,7 @@ UCS_TEST_P(test_memtype_cache_deferred_create, allocate_and_update_portions) {
         test_unknown_region_found(b);
         test_region_not_found(b);
 
-        /* add two protions of the region that shares one adderss in order
-         * to get them merged in a MemType Cache (the last address of the
-         * 1st protion is eqaul to the first address of the 2nd portion): */
-        /* - add the 1st portion */
-        memtype_cache_update(b.ptr(), b.size() / 2, b.mem_type());
-        test_ptr_found(b.ptr(), b.size() / 2, b.mem_type());
-        /* - add the 2nd portion */
-        memtype_cache_update(UCS_PTR_BYTE_OFFSET(b.ptr(), b.size() / 2 - 1),
-                             b.size() / 2 + 1, b.mem_type());
-        test_ptr_found(b.ptr(), b.size() / 2, b.mem_type());
-
-        /* check that able to find the entire region */ 
-        test_region_found(b);
+        test_adjacent_regions(b, 1);
```

Comment:
there is still code duplication between the tests which could be placed in test_adjacent_regions

Reply:
@yosefe introduced single function for 3 tests that can test adjacent regions when requested by a test thru parameter